packages/api/tests/api/contributions.test.ts

```diff
@@ -0,0 +1,163 @@
+    author: sampleUser as User,
+    type: 'CLOSED',
+    score: 1,
+    contributedAt: new Date('2011-01-02'),
+  },
+  {
+    nodeID: 'PR_T4C0B311',
+    description: 'yum',
+    author: sampleUser as User,
+    type: 'OPEN',
+    score: 123,
+    contributedAt: new Date('2011-01-03'),
+  },
+];
+
+const loggerSpy = jest.spyOn(logger, 'error').mockImplementation();
+const mockGetItems = jest.spyOn(Collection.prototype as Collection<Contribution>, 'getItems');
+
+describe('Contributions API GET route', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it('returns a 500 error when find throws an error', async () => {
+    const handler = testHandler(contributions, (req, _res, next) => {
+      req.user = sampleSignedInUser;
+      req.query = { userId: sampleUser.id };
+      next();
+    });
+
+    handler.entityManager.findOne.mockRejectedValueOnce(new Error(''));
+
+    const { text } = await handler.get('').expect(500);
+    expect(text).toEqual('There was an issue retrieving contributions');
+    expect(handler.entityManager.findOne).toBeCalledTimes(1);
+    expect(loggerSpy).toBeCalledTimes(1);
+  });
+
+  it('returns a 404 error when queried user cannot be found', async () => {
+    const handler = testHandler(contributions, (req, _res, next) => {
+      req.user = sampleSignedInUser;
+      req.query = { userId: sampleUser.id };
+      next();
+    });
+
+    handler.entityManager.findOne.mockResolvedValueOnce(null);
+    handler.entityManager.find.mockResolvedValueOnce(mockQueriedUserContributions);
+
+    await handler.get('').expect(404);
+    expect(handler.entityManager.findOne).toHaveBeenCalledTimes(1);
+    expect(handler.entityManager.find).toHaveBeenCalledTimes(0);
+  });
+
+  it('returns a 400 error when no id is entered in query parameters', async () => {
+    const handler = testHandler(contributions, (req, _res, next) => {
+      req.user = sampleSignedInUser;
+      next();
+    });
+
+    const { text } = await handler.get('').expect(400);
+    expect(text).toEqual("Query parameter 'userId' is required and must be a positive integer");
+  });
+
+  it('returns a 400 error when non-numeric userId passed in query parameters', async () => {
+    const mockQueryId = 'abc';
+
+    const handler = testHandler(contributions, (req, _res, next) => {
+      req.user = sampleSignedInUser;
+      req.query = { userId: mockQueryId };
+      next();
+    });
+
+    const { text } = await handler.get('').expect(400);
+    expect(text).toEqual("Query parameter 'userId' is required and must be a positive integer");
+  });
+
+  it("successfully returns the current user's contributions", async () => {
+    const handler = testHandler(contributions, (req, _res, next) => {
+      req.user = sampleSignedInUser;
+      req.query = { userId: sampleUser.id };
+      next();
+    });
+
+    // Deep copy made because the GET route will delete the author from each Contribution in the array
```

Comment:
One point of concern I have is whether this is a valid way to clone the mockQueriedUserContributions array in this situtation.

In the expect function of line 140, before I used this copy, I tried comparing the body to mockQueriedUserContributions and the only area where the expect failed was checking the contributedAt fields. The expect received the date as a string (2011-01-01T00:00:00.000Z in quotes) while the mockQueriedUserContributions array had it as a Date (just as 2011-01-01T00:00:00.000Z).

After making the copy using JSON.parse(JSON.stringify()), the expect worked.

Reply:
Yea you don't want to go comparing the same objects unless you're expecting the exact same object; the right thing to do here would be to remove your authors within the test and then compare OR use `expect.objectContaining` combined with `not` to make sure author ISN'T in the returned value